bindings/rust/evmc-vm/src/lib.rs

```diff
@@ -128,6 +363,144 @@ extern "C" fn release_result(result: *const ffi::evmc_result) {
     }
 }
 
+/*
+struct MyVM {}
+
+impl EVMCInstance for MyVM {
+  fn execute(...) {
+  }
+}
+
+evmc_create_vm!("myvm", "1.0", MyVM)
+
+*/
+
+//pub struct EvmcInstance(Box<ffi::evmc_instance>);
+/*
+pub struct EvmcInstance {
+    instance: ffi::evmc_instance,
+    executor: VMInstance
+}
+
+pub trait VMInstance {
+  fn execute(instance: EvmcInstance, host: ffi::evmc_context, revision: ffi::evmc_revision, msg: ffi::evmc_message, code: &[u8]) -> ffi::evmc_result;
+//  pub fn set_option();
+//  pub fn set_tracer();
+}
+*/
+
+#[macro_export]
+macro_rules! evmc_create_vm {
+    ($__vm:ident, $__version:expr) => {
+        paste::item! {
+            static [<$__vm _NAME>]: &'static str = stringify!($__vm);
+            static [<$__vm _VERSION>]: &'static str = $__version;
+        }
+
+        paste::item! {
```

Comment:
Should have here a large comment explaining this will lay out memory in a format acceptable to C (`repr(C)`) and that's why we can include both data structures here.

Reply:
will do